python/tskit/tables.py

```diff
@@ -1367,22 +1371,23 @@ def _text_header_and_rows(self):
         site = self.site
         node = self.node
         parent = self.parent
+        time = self.time
         derived_state = util.unpack_strings(
             self.derived_state, self.derived_state_offset
         )
         metadata = util.unpack_bytes(self.metadata, self.metadata_offset)
-        headers = ("id", "site", "node", "derived_state", "parent", "metadata")
+        headers = ("id", "site", "node", "time", "derived_state", "parent", "metadata")
         rows = []
         for j in range(self.num_rows):
             md = base64.b64encode(metadata[j]).decode("utf8")
             rows.append(
-                "{}\t{}\t{}\t{}\t{}\t{}".format(
-                    j, site[j], node[j], derived_state[j], parent[j], md
+                "{}\t{}\t{}\t{}\t{}\t{}\t{}".format(
+                    j, site[j], node[j], time[j], derived_state[j], parent[j], md
                 ).split("\t")
             )
         return headers, rows
 
-    def add_row(self, site, node, derived_state, parent=-1, metadata=None):
+    def add_row(self, site, node, time, derived_state, parent=-1, metadata=None):
```

Comment:
I'd have been inclined to put this after derived_state as an optional value with default 0 to avoid some breakage. I agree this is conceptually the right place for it though. Since you've gone through the pain of updating the test suite it's probably best to keep it as it is. I doubt we'll break *that* much downstream code...